src/JitInterface/src/TypesDebugInfo/PrimitiveTypeDescriptor.cs

```diff
@@ -0,0 +1,253 @@
+            T_64PCPLX32 = 0x0650,   // 64 bit pointer to 32 bit complex
+
+            //  64 bit complex types
+
+            T_CPLX64 = 0x0051,   // 64 bit complex
+            T_32PCPLX64 = 0x0451,   // 32 bit pointer to 64 bit complex
+            T_64PCPLX64 = 0x0651,   // 64 bit pointer to 64 bit complex
+
+            //  80 bit complex types
+
+            T_CPLX80 = 0x0052,   // 80 bit complex
+            T_32PCPLX80 = 0x0452,   // 32 bit pointer to 80 bit complex
+            T_64PCPLX80 = 0x0652,   // 64 bit pointer to 80 bit complex
+
+            //  128 bit complex types
+
+            T_CPLX128 = 0x0053,   // 128 bit complex
+            T_32PCPLX128 = 0x0453,   // 32 bit pointer to 128 bit complex
+            T_64PCPLX128 = 0x0653,   // 64 bit pointer to 128 bit complex
+
+            //  boolean types
+
+            T_BOOL08 = 0x0030,   // 8 bit boolean
+            T_32PBOOL08 = 0x0430,   // 32 bit pointer to 8 bit boolean
+            T_64PBOOL08 = 0x0630,   // 64 bit pointer to 8 bit boolean
+
+            T_BOOL16 = 0x0031,   // 16 bit boolean
+            T_32PBOOL16 = 0x0431,   // 32 bit pointer to 18 bit boolean
+            T_64PBOOL16 = 0x0631,   // 64 bit pointer to 18 bit boolean
+
+            T_BOOL32 = 0x0032,   // 32 bit boolean
+            T_32PBOOL32 = 0x0432,   // 32 bit pointer to 32 bit boolean
+            T_64PBOOL32 = 0x0632,   // 64 bit pointer to 32 bit boolean
+
+            T_BOOL64 = 0x0033,   // 64 bit boolean
+            T_32PBOOL64 = 0x0433,   // 32 bit pointer to 64 bit boolean
+            T_64PBOOL64 = 0x0633,   // 64 bit pointer to 64 bit boolean
+        };
+
+
+        public static uint GetPrimitiveTypeIndex(TypeDesc typeDescriptor)
+        {
+            Debug.Assert(typeDescriptor.IsPrimitive, "it is not primitive type");
+            switch (typeDescriptor.Category)
+            {
+                case TypeFlags.Void:
+                    return (uint)TYPE_ENUM.T_VOID;
+                case TypeFlags.Boolean:
+                    return (uint)TYPE_ENUM.T_BOOL08;
+                case TypeFlags.Char:
+                    return (uint)TYPE_ENUM.T_WCHAR;
+                case TypeFlags.SByte:
+                    return (uint)TYPE_ENUM.T_INT1;
+                case TypeFlags.Byte:
+                    return (uint)TYPE_ENUM.T_UINT1;
+                case TypeFlags.Int16:
+                    return (uint)TYPE_ENUM.T_INT2;
+                case TypeFlags.UInt16:
+                    return (uint)TYPE_ENUM.T_UINT2;
+                case TypeFlags.Int32:
+                    return (uint)TYPE_ENUM.T_INT4;
+                case TypeFlags.UInt32:
+                    return (uint)TYPE_ENUM.T_UINT4;
+                case TypeFlags.Int64:
+                    return (uint)TYPE_ENUM.T_INT8;
+                case TypeFlags.UInt64:
+                    return (uint)TYPE_ENUM.T_UINT8;
+                case TypeFlags.IntPtr:
+                    return (uint)TYPE_ENUM.T_32PLONG;
```

Comment:
```
T_32PLONG = 0x0412,   // 32 bit pointer to 32 bit signed 
```

That doesn't seem right, especially since we only support x64 right now.

`T_32PVOID` and `T_64PVOID` depending on `typeDescriptor.Context.Target.PointerSize` would be more fitting.